collector.go

```diff
@@ -153,14 +155,44 @@ var (
 	)
 )
 
+func freeipmiConfig(driver, user, password string) string {
+	return fmt.Sprintf(`
+driver-type %s
+privilege-level admin
+username %s
+password %s
+	`, driver, user, password)
+}
+
+func freeipmiTempConfigFile(driver, user, password string) (string, error) {
+	content := []byte(freeipmiConfig(driver, user, password))
+	tmpfile, err := ioutil.TempFile("", "ipmi-exporter")
+	if err != nil {
+		return "", err
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		defer os.Remove(tmpfile.Name())
```

Comment:
Adding a new deferred function right before exiting appears to be unnecessary. Just call os.Remove() directory?

Reply:
Good point